src/control/server/mgmt_system.go

```diff
@@ -1264,20 +1208,38 @@ func (svc *mgmtSvc) SystemDrain(ctx context.Context, req *mgmtpb.SystemDrainReq)
 		return nil, err
 	}
 
+	if len(poolIDs) != len(poolRanks) {
+		return nil, errors.New("nr poolIDs should be equal to poolRanks keys")
+	}
+	if len(poolIDs) == 0 {
+		return nil, errors.New("no pool-ranks found to operate on with request params")
+	}
+
 	// Generate results from dRPC calls.
-	var opCall poolRankOpSig = drainPoolRanks
-	if req.Reint {
-		opCall = reintPoolRanks
+	var apiCall poolRanksOpSig = control.PoolDrain
+	if pbReq.Reint {
+		apiCall = control.PoolReintegrate
 	}
-	results, err := svc.getPoolRankResults(ctx, req.Sys, poolIDs, poolRanks, opCall)
+	resps, err := svc.getPoolRanksResps(ctx, pbReq.Sys, poolIDs, poolRanks, apiCall)
 	if err != nil {
 		return nil, err
 	}
 
-	return &mgmtpb.SystemDrainResp{
-		Reint:   req.Reint,
-		Results: results,
-	}, nil
+	if len(resps) == 0 {
+		return nil, errors.New("no pool-ranks responses received")
+	}
+	if len(resps) != len(poolIDs) {
+		return nil, errors.Errorf("unexpected number of pool-ranks responses received, "+
+			"want %d got %d", len(poolIDs), len(resps))
+	}
```

Comment:
IMO should log and return the surprising response anyway

Reply:
done

Reply:
Doesn't look like this one made it into the update. I think it's fine to include in a follow-on though.